### PR TITLE
feat: optionally show empty dropdown

### PR DESCRIPTION
**Pull request by fmehta-eightfold**
## SUMMARY:
In somecases empty dropdown is not required in the UI

## JIRA TASK (Eightfold Employees Only):
ENG-39498

## CHANGE TYPE:

- [ ] Bugfix Pull Request
- [x] Feature Pull Request

## TEST COVERAGE:

- [ ] Tests for this change already exist
- [ ] I have added unittests for this change

## TEST PLAN:
Passing `showEmptyDropdown = false` will not show the dropdown when `options.length === 0`
**Before**
image
**After**
image

### CODE REVIEW

File: src/components/Select/Select.tsx [hunk 1]
```diff
@@ -535,7 +536,9 @@ export const Select: FC<SelectProps> = React.forwardRef(
             onVisibleChange={(isVisible) => setDropdownVisibility(isVisible)}
             overlay={isLoading ? spinner : <OptionMenu options={options} />}
             showDropdown={showDropdown}
-            visible={dropdownVisible}
+            visible={
+              dropdownVisible && (showEmptyDropdown || options.length > 0)
```
Comment (dkilgore-eightfold) [inline src/components/Select/Select.tsx:540]: Does this account for the type-ahead scenario and still show the 'No match found' result? Never showing makes sense for on click with no input, but should provide some information with a partially input text.

**Comment by slee-eightfold** [comment]
needs_sandbox

**Comment by slee-eightfold** [comment]
@fmehta-eightfold How do I test the changes on sandbox? I need to run it on PCS.

File: src/components/Select/Select.tsx (hunk 1 above)
Comment (fmehta-eightfold) [reply src/components/Select/Select.tsx]: You are right, this is done to handle the case of `click with no input`, and the `No match found` message will be shown with partially input text.

**Comment by fmehta-eightfold** [comment]
> @fmehta-eightfold How do I test the changes on sandbox? I need to run it on PCS.

Refer vscode/pull/35681 on how to test this.

**Review by slee-eightfold — APPROVED** [review]
Tested in PCS Apply Form with the following conditions and is working as expected:
- empty options & showEmptyDropdown == true
Screenshot 2023-02-01 at 10 02 50 PM

- empty options & showEmptyDropdown == false
Screenshot 2023-02-01 at 10 03 11 PM

- non-empty options && showEmptyDropdown == false
Screenshot 2023-02-01 at 10 03 46 PM

**Pull request merged**